src/Build/Evaluation/Evaluator.cs

```diff
@@ -1830,38 +1840,47 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
 
                         _evaluationLoggingContext.LogBuildEvent(eventArgs);
 
-                        projects = new List<ProjectRootElement>();
-
                         return;
                     }
 
                     ProjectErrorUtilities.ThrowInvalidProject(importElement.SdkLocation, "CouldNotResolveSdk", sdkReference.ToString());
                 }
 
-                if (sdkResult.Path == null)
-                {
-                    projects = new List<ProjectRootElement>();
-                }
-                else
+                if (sdkResult.Path != null)
                 {
                     ExpandAndLoadImportsFromUnescapedImportExpression(directoryOfImportingFile, importElement, Path.Combine(sdkResult.Path, project),
                         throwOnFileNotExistsError, out projects);
 
+                    if (projects?.Count > 0)
+                    {
+                        projects = new List<ProjectRootElement>(projects);
+                    }
+
                     if (sdkResult.AdditionalPaths != null)
                     {
+                        if (projects == null && sdkResult.AdditionalPaths.Count > 0)
+                        {
+                            projects = new List<ProjectRootElement>();
+                        }
```

Comment:
nit:

```suggestion
                    if (sdkResult.AdditionalPaths?.Count > 0)
                    {
                        projects ??= new List<ProjectRootElement>();
```

Or maybe even:
```suggestion
                    if (sdkResult.AdditionalPaths?.Count > 0)
                    {
```

and new up the `List` lazily when actually adding to it a few lines below.

Reply:
Oh boy! That's an embarrassing refactoring mistake. I'm starting to have selective blindness in this part of code. It's scary that all tests passed... It should be fixed now.